test/test-applications/integrations/TestApplication.ContinuousProfiler/Exporter/SampleNativeFormatParser.cs

```diff
@@ -0,0 +1,308 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Text;
+
+namespace TestApplication.ContinuousProfiler;
+
+/// <summary>
+/// Parser the native code's pause-time-optimized format.
+/// </summary>
+internal static class SampleNativeFormatParser
+{
+    private static readonly UnicodeEncoding UnicodeEncoding = new();
+
+    /// <summary>
+    /// Parses the thread sample batch.
+    /// </summary>
+    /// <param name="buffer">byte array containing native thread samples format data</param>
+    /// <param name="read">how much of the buffer is actually used</param>
+    internal static List<ThreadSample>? ParseThreadSamples(byte[] buffer, int read)
+    {
+        uint batchThreadIndex = 0;
+        var samples = new List<ThreadSample>();
+        long sampleStartMillis = 0;
+
+        var position = 0;
+
+        // common for samples in a batch
+        var codeDictionary = new Dictionary<int, string>();
+
+        try
+        {
+            while (position < read)
+            {
+                var operationCode = buffer[position];
+                position++;
+                if (operationCode == OpCodes.StartBatch)
+                {
+                    var version = ReadInt(buffer, ref position);
+                    if (version != 1)
+                    {
+                        return null; // not able to parse
+                    }
+
+                    sampleStartMillis = ReadInt64(buffer, ref position);
+
+                    /* if (IsLogLevelDebugEnabled)
```

Comment:
commented code (here and below), if worth adding TODO comment seems appropriate. Possibly a separate issue also for the future to track.

Reply:
#3205 and 77f6a6048795efa5c50e2c4eb962c40d4e895b32